### DISCUSSION TITLE
Spoke export no shadows anymore

### DISCUSSION THREAD

**Discussion by ArneKossmann** [mozilla/hubs#3077] [Spoke Questions] [answered]
Hey,
actually I'm working on a custom client project on AWS(installed on July2020), without any problems so far.
But since a few days our designer can not export shadows via spoke anymore.
In old scenes the shadows are still visible until we re-published them.

_Edit:_
I've also undeployed our custom client to check if there's any problem. But we got the same results.
Should we update our spoke on AWS somehow?

![Schatten Test _ Scene Editor - 1](https://user-images.githubusercontent.com/72018287/94424710-f3a93780-018a-11eb-9c2f-91408845e7f2.jpg)
![Schatten Test _ Scene Editor - 2](https://user-images.githubusercontent.com/72018287/94424717-f60b9180-018a-11eb-8dc5-ee296ca1f9df.jpg)
Thanks in advance,
Arne.

**Comment by cvanmeurs** [comment]
Sorry, this issue seems to have slipped through the cracks - looks like it's been a while since you posted this.  Have you resolved the issue since then?  If you're still having trouble, I'd be happy to take a look at your scene for you and see if I can figure out what's going on.

**Reply by ArneKossmann** [reply]
Hey!
Unfortunately I was not really able to fix this issue. But I did some more tests and the shadow-problem only appears with the directional-light.
Here's a test-scene in Spoke...
![2020-12-04 11_04_41-Window](https://user-images.githubusercontent.com/72018287/101152183-ff015f00-3622-11eb-8bc7-0fc31536856d.png)
...as you can see. I placed the light a little bit off center.
And here is the scene in hubs...
![2020-12-04 11_06_21-Window](https://user-images.githubusercontent.com/72018287/101152368-40920a00-3623-11eb-8f54-d2b010376c4b.png)
...the shadow is cropped off. It looks like the shadow camera frustum is too small.
[https://threejs.org/docs/index.html#api/en/cameras/OrthographicCamera](https://threejs.org/docs/index.html#api/en/cameras/OrthographicCamera)
So I've found this function "resizeShadowCameraFrustum" in both sources.
[https://github.com/mozilla/Spoke/blob/master/src/editor/utils/resizeShadowCameraFrustum.js](https://github.com/mozilla/Spoke/blob/master/src/editor/utils/resizeShadowCameraFrustum.js)
[https://github.com/mozilla/hubs/blob/master/src/systems/shadow-system.js](https://github.com/mozilla/hubs/blob/master/src/systems/shadow-system.js)
But I cound't figure out why it works in Spoke and not in Hubs. 
At the moment I use the light scaling instead as a workaround.

Oh and here's the link to the remixable test-scene...
[https://hubs.mozilla.com/scenes/WYb5Eqy/shadow-test](https://hubs.mozilla.com/scenes/WYb5Eqy/shadow-test)
[https://hubs.mozilla.com/KZ6eMtD/shadow-test/](https://hubs.mozilla.com/KZ6eMtD/shadow-test/)
(I'm not sure how to share scenes :/ )

Thanks for your help!

**Comment by cvanmeurs** [comment]
@robertlong Hey Robert, do you know what could be going on here?  It's a simple setup and I'm able to reproduce it - shadows look correct in Spoke but are not in Hubs...

**Reply by j-conrad** [reply]
I also reproduced it using the same light transform settings in the screenshot. I seem to recall some change that occurred to shadow frustum stuff a while back, but I forget exactly.

**Comment by cvanmeurs** [comment]
Well this is strange - I added a second object to the scene and suddenly both have correct shadows in Hubs.

**Comment by cvanmeurs** [comment, marked as answer]
Yeah, duplicate that floating floor piece and move it over and publish again - works.

**Reply by j-conrad** [reply]
Same. But not if you move it upwards.

**Reply by cvanmeurs** [reply]
Mine works.  But I didn't move it *straight* up...

**Reply by cvanmeurs** [reply]
![shadows](https://user-images.githubusercontent.com/63256260/101225183-a3949880-3645-11eb-9d24-febc10d86f1b.png)

**Reply by j-conrad** [reply]
Looks like the duplicate was a red herring of sorts...    I can get the shadow to draw correctly if I move a single floor piece away from the origin.

**Reply by cvanmeurs** [reply]
Yeah you're right; if you move it straight up it's still borked.  WEIRD

**Reply by cvanmeurs** [reply]
I get the same results if I move the 2 stacked pieces laterally away from the origin - no shadows at all, in fact.  Not even cast onto the lower floating platform.
